includes/abstracts/llms-abstract-processor-user-engagement-sync.php

```diff
@@ -0,0 +1,304 @@
+			sprintf(
+				'awarded %1$ss bulk sync dispatched for the %1$s template %2$s (#%3$d)',
+				$this->engagement_type,
+				get_the_title( $engagement_template_id ),
+				$engagement_template_id
+			)
+		);
+
+		/**
+		 * Filter the query arguments used when retrieving the awarded engagements to sync.
+		 *
+		 * The dynamic portion of the hook name,
+		 * {@see LLMS_Abstract_Processor_User_Engagement_Sync::$engagement_type `$this->engagement_type`},
+		 * refers to the engagement type, either 'achievement' or 'certificate'.
+		 *
+		 * @since [version]
+		 *
+		 * @param array $args Query arguments passed to LLMS_Awards_Query.
+		 */
+		$args = apply_filters(
+			"llms_processor_sync_awarded_{$this->engagement_type}s_query_args",
+			array(
+				'templates' => $engagement_template_id,
+				'per_page'  => 20,
+				'page'      => 1,
+				'status'    => array(
+					'publish',
+					'future',
+				),
+				'type'      => $this->engagement_type,
+			)
+		);
+
+		$query = new LLMS_Awards_Query( $args );
+
+		if ( ! $query->get_found_results() ) {
+			return;
+		}
+
+		while ( $args['page'] <= $query->get_max_pages() ) {
+			$this->push_to_queue(
+				array(
+					'query_args' => $args,
+				)
+			);
+
+			$args['page'] ++;
+		}
+
+		// Save queue and dispatch the process.
+		$this->save()->dispatch();
+	}
+
+	/**
+	 * Initializer.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	protected function init() {
+
+		// For the cron.
+		add_action( $this->schedule_hook, array( $this, 'dispatch_sync' ), 10, 1 );
+
+		// For LifterLMS actions which trigger bulk enrollment.
+		$this->actions = array(
+			"llms_do_awarded_{$this->engagement_type}s_bulk_sync" => array(
+				'arguments' => 1,
+				/** @see LLMS_Abstract_Processor_User_Engagement_Sync::schedule_sync() */
+				'callback'  => 'schedule_sync',
+				'priority'  => 10,
+			),
+		);
+	}
+
+	/**
+	 * Perform actions when the process is completed.
+	 *
+	 * @since [version]
+	 *
+	 * @param array $args Array of processing data.
+	 * @return void
+	 */
+	private function process_completed( $args ) {
+
+		$this->log(
+			sprintf(
+				'awarded %1$s bulk sync completed for the %1$s template %2$s (#%3$d)',
+				$this->engagement_type,
+				get_the_title( $args['query_args']['templates'] ),
+				$args['query_args']['templates']
+			)
+		);
+
+		$this->clear_notices( $args['query_args']['templates'] );
+
+		LLMS_Admin_Notices::add_notice(
+			sprintf( 'awarded-%1$ss-sync-%2$d-done', $this->engagement_type, $args['query_args']['templates'] ),
+			sprintf(
+				/* translators: 1: plural uppercase first letter awarded engagement type, 2: opening anchor tag that links to the engagement template, 3: engagement template name, 4: engagement template ID, 5: closing anchor tag, 6: plural lowercase awarded engagement type */
+				__( '%1$s sync completed for the template %2$s%3$s (#%4$d)%5$s.', 'lifterlms' ),
+				$this->get_engagement_type_name( self::PLURAL, self::CASE_UPPER_FIRST, self::AWARDED ),
```

Comment:
The completion notice for both certificates & achievements when bulk syncing from template -> awarded versions resulted in "Unknown Engagement Type sync completed...". I'm not sure where the problem lies exactly but I saw it on both of them